src/cargo/util/toml/mod.rs

```diff
@@ -2550,10 +2550,10 @@ pub fn validate_profile(
     }
 
     if let Some(panic) = &root.panic {
-        if panic != "unwind" && panic != "abort" {
+        if panic != "unwind" && panic != "abort" && panic != "immediate-abort" {
```

Comment:
Not sure if we want to make it a warning or  configurable lint in the future. It seems too restrictive. Anyway, it is worth its own separate topic.